pkg/common/health/health.go

```diff
@@ -0,0 +1,95 @@
+package health
+
+import (
+	"context"
+	"net/http"
+	"sync"
+
+	"github.com/InVisionApp/go-health"
+	"github.com/InVisionApp/go-health/handlers"
+	"github.com/sirupsen/logrus"
+)
+
+// health.Checker is responsible for running health checks and serving the healthcheck HTTP paths
+type Checker struct {
+	config Config
+
+	server *http.Server
+
+	hc *health.Health
+
+	log logrus.FieldLogger
+}
+
+func hello(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "text/plain")
+	w.WriteHeader(http.StatusOK)
+}
+
+func NewChecker(config Config, log logrus.FieldLogger) *Checker {
+	hc := health.New()
+
+	var server http.Server
+	// Start HTTP server if address is configured
+	if config.Address != "" {
+		handler := http.NewServeMux()
+
+		handler.HandleFunc(config.getHealthPath(), handlers.NewJSONHandlerFunc(hc, nil))
+		handler.HandleFunc(config.getLivePath(), hello)
```

Comment:
Based on this, I assume that Health == k8s readiness and Live == k8s liveness?

Perhaps we should converge on the k8s naming for these two checks to avoid confusion... same for default paths e.g. /ready ?

Reply:
Yes, I'm happy to use /ready and /live (and corresponding configurable names) to be consistent with k8s.

Reply:
Done, code uses ready and live now.